packages/pyright-internal/src/common/positionUtils.ts

```diff
@@ -8,7 +8,7 @@
  * line/column positions.
  */
 
-import { TokenizerOutput } from '../parser/tokenizer';
+import { ParseResults } from '../parser/parser';
```

Comment:
This creates an import circularity. The parser.ts file already imports positionUtils.ts. While TypeScript is able to deal with this circularity, it's something I try very hard to avoid because it's a sign of incorrect architectural layering. In general, "util" files should be leaf nodes (or close to leaf nodes) in the import tree.

Reply:
In this case, the function getLineEndOffset should not accept a ParseResults. It should accept the tokenizerOutput and text lines as separate parameters.